modules/clients/main.tf

```diff
@@ -102,6 +106,26 @@ resource "azurerm_network_interface_security_group_association" "private" {
   network_security_group_id = var.sg_id
 }
 
+data "azurerm_user_assigned_identity" "this" {
+  count               = var.vm_identity_name != "" ? 1 : 0
+  name                = var.vm_identity_name
+  resource_group_name = data.azurerm_resource_group.rg.name
+}
+
+resource "azurerm_user_assigned_identity" "this" {
+  count               = var.vm_identity_name == "" ? 1 : 0
+  location            = data.azurerm_resource_group.rg.location
+  name                = "${var.clients_name}-identity"
+  resource_group_name = data.azurerm_resource_group.rg.name
+}
+
+resource "azurerm_role_assignment" "reader" {
+  count                = var.vm_identity_name == "" ? 1 : 0
+  scope                = data.azurerm_resource_group.rg.id
+  role_definition_name = "Reader"
```

Comment:
why do we need this? for fetching the ips of backends? can we have more specific role?